hazelcast/src/main/java/com/hazelcast/sql/impl/SqlServiceImpl.java

```diff
@@ -203,6 +204,10 @@ public SqlResult execute(@Nonnull SqlStatement statement, SqlSecurityContext sec
         Preconditions.checkNotNull(statement, "Query cannot be null");
 
         try {
+            if (nodeEngine.getClusterService().getClusterVersion().isLessThan(Versions.V5_0)) {
+                throw QueryException.error("SQL queries cannot be executed on clusters with version less than 5.0");
```

Comment:
I'd change this text. It's a bit misleading because there was SQL support in 4.2.
```suggestion
                throw QueryException.error("SQL queries cannot be executed until the cluster fully updates to 5.0");
```